File: app/schemas/product.py
'''
商品数据模型 这份代码是连接“数据层”和“算法层”的桥梁
'''
from pydantic import BaseModel, Field


class Sku(BaseModel):
    sku_id: str
    properties: dict[str, str] = Field(default_factory=dict)
    price: float


class FaqItem(BaseModel):
    question: str
    answer: str


class ReviewItem(BaseModel):
    nickname: str
    rating: int  # 1-5
    content: str


class RagKnowledge(BaseModel):
    marketing_description: str = ""
    official_faq: list[FaqItem] = Field(default_factory=list)
    user_reviews: list[ReviewItem] = Field(default_factory=list)


class Product(BaseModel):
    product_id: str
    title: str
    brand: str
    category: str  # 美妆护肤 / 数码电子 / 服饰运动 / 食品饮料
    sub_category: str = ""
    base_price: float
    image_path: str = ""  # 数据集内相对路径
    skus: list[Sku] = Field(default_factory=list)
    rag_knowledge: RagKnowledge | None = None

File: android-client/app/src/main/java/com/omnicart/agent/core/model/Product.kt
package com.omnicart.agent.core.model

import com.google.gson.annotations.SerializedName

data class Product(
    @SerializedName("product_id")
    val productId: String = "",
    @SerializedName("title")
    val title: String = "",
    @SerializedName("brand")
    val brand: String = "",
    @SerializedName("category")
    val category: String = "",
    @SerializedName("sub_category")
    val subCategory: String = "",
    @SerializedName("price")
    val price: Double = 0.0,
    @SerializedName("image_urls")
    val imageUrls: List<String> = emptyList(),
    @SerializedName("skus")
    val skus: List<Sku>? = null,
    @SerializedName("rag_knowledge")
    val ragKnowledge: RagKnowledge? = null,
    @SerializedName("description")
    val description: String = ""
)

data class Sku(
    @SerializedName("sku_id")
    val skuId: String = "",
    @SerializedName("properties")
    val properties: Map<String, String>? = null,
    @SerializedName("price")
    val price: Double = 0.0
)

data class RagKnowledge(
    @SerializedName("marketing_description")
    val marketingDescription: String = "",
    @SerializedName("official_faq")
    val officialFaq: List<FaqItem>? = null,
    @SerializedName("user_reviews")
    val userReviews: List<ReviewItem>? = null
)

data class FaqItem(
    @SerializedName("question")
    val question: String = "",
    @SerializedName("answer")
    val answer: String = ""
)

data class ReviewItem(
    @SerializedName("nickname")
    val nickname: String = "",
    @SerializedName("rating")
    val rating: Int = 0,
    @SerializedName("content")
    val content: String = ""
)
